src/TensorFlowNET.Keras/Engine/Layer.cs

```diff
@@ -134,6 +134,30 @@ public virtual List<IVariableV1> Weights
             }
         }
 
+        public virtual void set_weights(List<NDArray> weights)
+        {
+            if (Weights.Count() != weights.Count()) throw new ValueError(
+                                            $"You called `set_weights` on layer \"{this.name}\"" +
+                                            $"with a weight list of length {len(weights)}, but the layer was " +
+                                            $"expecting {len(Weights)} weights.");
+            for (int i = 0; i < weights.Count(); i++)
+            {
+                if (weights[i].shape != Weights[i].shape)
```

Comment:
The condition seems to be too strict. For example, shape A is (-1, 10), shape B is (8, 10) is also compatible. (refer to function `is_compatible_with` in tensorflow).